docs/src/ci.md

```diff
@@ -0,0 +1,143 @@
+
+- **Unit tests**: tests which check the functionality of the modified (sub) package. Those tests can either run standalone or use the functionality of other (third-party) packages.
+- **Integration tests**: tests which check the interoperatebility of the modified package with all sub-packages. For example, if package `A` is dependent on package `B` and you change package `B`, the integration test will check, if package `A` still works with the new version of package `B`.
+
+The CI will first execute the unit tests. Then, if all unit tests are passed, the integration tests will be started. 
+
+Our CI uses the [GitLab CI](https://docs.gitlab.com/ee/ci/) because it allows us to use CI resources provided by [HIFIS](https://www.hifis.net/), where we can strong CPU runners and special runners, like Nvidia and AMD GPUs.
+
+
+# Unit Test for CI Users
+
+The unit tests are automatically triggered, if you open a pull request, which target the `main` or `dev` branch. Before the tests start, the CI setup the test environment. Therefore it takes the `Project.toml` of a project and add the latest development version of each QED dependency. Other dependencies will be regular resolved by the Julia Package manager. Afterwards the unit tests will be executed.
+
+You can also modify, which version of a QED dependency should be used. For example you need to test your code with a function, which is not merged in the development branch yet. Therefore you need to add specific line to your commit message with the following format:
+
+```
+CI_UNIT_PKG_URL_<dep_name>: https://github.com/<user>/<dep_name>#<commit_hash>
+```
+
+You find the `<dep_name>` name in the `Project.toml` of the dependent project. For an example let's assume the name of the dependent package is `depLibrary`. The url of the fork is https://github.com/user/depLibrary.jl and the required feature has the commit sha `45a753b`.
+
+```
+This commit extends function foo with a new
+function argument.
+
+The function argument is required to control
+the new functionality.
+
+If you pass a 0, it has a special meaning.
+
+CI_UNIT_PKG_URL_depLibrary: https://github.com/user/depLibrary.jl#45a753b
+```
+
+It is also possible to set a custom URL for more than one package. Simple add an additional line with the shape of `CI_UNIT_PKG_URL_<dep_name>: https://github.com/<user>/<dep_name>#<commit_hash>` to the commit message.
+
+!!! note
+
+    You don't need to add a new commit to set custom URLs. You can modify the commit message with `git commit --amend` and force push to the branch. This also starts the CI pipeline again.
+
+There is a last job, which checks if lines starting with `CI_UNIT_PKG_URL_` are exist in the commit message of the last commit. If they are exist, the test fails. This is required, because otherwise the merge code would be depend on non merged changes in sub packages and would be not compilable.
```

Comment:
are exist -> exist
last commit -> latest commit 

If they are exist, the test fails. -> If so, the unit test will fail. 

the merge code -> the merged code
would be depend -> would depend 
would be not compatible -> would be non-compatible